webapp/channels/src/components/interactive_dialog/interactive_dialog.tsx

```diff
@@ -98,13 +110,11 @@ export default class InteractiveDialog extends React.PureComponent {
             callback_id: callbackId,
             state,
             submission: values,
-        };
+        } as unknown as DialogSubmission;
 
         this.setState({submitting: true});
 
-        const {data} = await this.props.actions.submitInteractiveDialog(
-            dialog,
-        );
+        const {data}: any = await this.props.actions.submitInteractiveDialog(dialog)(store.dispatch, store.getState);
```

Comment:
I think this is not correct, we should be calling here this function and it should be already "connected" so we don't need to pass the dispatch or something like that. I think this is broken.

Reply:
Yess! even i thought so ! but according to the type definition of `submitInteractiveDiaog` it returns a `ActionFunc`! thats why i did like that!